packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -157,6 +158,7 @@ export function useInnerBlocksProps( props = {}, options = {} ) {
 
 	const ref = useMergeRefs( [
 		props.ref,
+		useInBetweenClickRedirect(),
```

Comment:
Ideally the block list root should just be reusing the inner blocks hooks. See #29895.